_posts/2021-03-02-introducing-new-window-features.md

```diff
@@ -0,0 +1,317 @@
+those encrypted scrolls to give you a new feature that you didn't even know you
+needed!
+
+Recently, there was a push in Trino to extend support for window functions.
+In this post, we are explaining the trivia of window functions and describing a
+couple of our recent additions. If "window" doesn't sound familiar, read on.
+Already a window expert? Skip to [what's new](#new features).
+
+Window is the structure you run your window function `OVER`. It has three
+components:
+- partitioning,
+- ordering,
+- frame.
+
+You use partitioning to break your input data into independent chunks. Ordering
+is to order rows within the partition. And frame is a kind of “sliding window”.
+For every processed row, the frame embraces a certain portion of the sorted
+partition. Your window function processes this portion and yields the result
+for the row.
+
+A “running average” is one simple example:
+
+```
+SELECT avg(totalprice) OVER (PARTITION BY custkey ORDER BY orderdate
+   ROWS BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW)
+FROM orders
+```
+
+For a particular customer identified by `custkey`, it sorts their orders by
+date and computes a sequence of average prices since the beginning up to each
+consecutive entry. The window frame for a row embraces all rows from the
+start up to and including that row.
+
+According to ISO SQL, there are 3 ways to specify the frame. The first way is
+`ROWS` (like in the example). With `ROWS`, you can specify frame bounds by a
+physical offset from the current row. While `ROWS BETWEEN
+UNBOUNDED PRECEDING AND CURRENT ROW` means “between the beginning of the
+partition and the current row”, you can also specify precisely where the frame
+should start and end, like this: `ROWS BETWEEN 10 PRECEDING AND 5 FOLLOWING`.
+
+`RANGE` is a more complicated way of defining frame on ordered data. It does
+not rely on physical offset (in rows) but on logical offset (in value). That
+is, the frame includes rows where the value is within certain range from the
+value in the current row.
+
+Until recently, Trino only supported `RANGE` in limited cases.
+You could use `RANGE UNBOUNDED PRECEDING`, `CURRENT ROW` and `UNBOUNDED
+FOLLOWING`. As easy as it seems, `UNBOUNDED PRECEDING` and `UNBOUNDED FOLLOWING`
+includes all rows until the partition bound. `CURRENT ROW` is more tricky. It
+includes all rows where values are the same as in the current row. We call
+them a _peer group_.
+
+It's time to introduce the first of Trino's new additions:
+
+### <a name="new features"/> Full support for frame type RANGE
+
+Since version 346, it is possible to specify `RANGE` with an offset value. The
+frame includes all rows whose value is within this range from the current row.
+
+Let's modify our example:
+
+```
+SELECT avg(totalprice) OVER (PARTITION BY custkey ORDER BY orderdate
+   RANGE BETWEEN interval '1' month PRECEDING AND CURRENT ROW)
+FROM orders
+```
+
+Now, we don't always get the average price since the beginning, but the average
```

Comment:
Not sure I understand what "we don't always get the average price since the beginning" is talking about